File: 剑指Offer/63.股票的最大利润.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        if (prices.size() == 0) return 0;
        int low = prices[0];
        int maxProfit = 0;
        for (int i = 1; i < prices.size(); i++) {
            maxProfit = max(maxProfit, prices[i] - low);
            low = min(low, prices[i]);
        }
        return maxProfit;
    }
};

File: 剑指Offer/63.股票的最大利润.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if len(prices) == 0:
            return 0
        
        low = prices[0]
        maxProfit = 0
        for i in range(1, len(prices)):
            maxProfit = max(maxProfit, prices[i] - low)
            low = min(low, prices[i])
        
        return maxProfit